src/entrypoint/graphql-admin-server.ts

```diff
@@ -0,0 +1,187 @@
+import { ApolloServer } from "apollo-server-express"
+import dotenv from "dotenv"
+import express from "express"
+import expressJwt from "express-jwt"
+import { applyMiddleware } from "graphql-middleware"
+import { and, rule, shield } from "graphql-shield"
+import mongoose from "mongoose"
+import pino from "pino"
+import PinoHttp from "pino-http"
+import { v4 as uuidv4 } from "uuid"
+
+import { yamlConfig } from "../config"
+import { AuthorizationError, IPBlacklistedError } from "../error"
+import { baseLogger } from "../logger"
+import { redis } from "../redis"
+import { User } from "../schema"
+import { setupMongoConnection } from "../mongodb"
+import helmet from "helmet"
+import { isDev, updateIPDetails, isIPBlacklisted } from "../utils"
+import { WalletFactory } from "../walletFactory"
+import { gqlAdminSchema } from "../graphql"
+
+dotenv.config()
+
+const graphqlLogger = baseLogger.child({ module: "graphql" })
+
+// TODO: extract common server configs
+
+const pino_http = PinoHttp({
+  logger: graphqlLogger,
+  wrapSerializers: false,
+
+  // Define custom serializers
+  serializers: {
+    err: pino.stdSerializers.err,
+    req: pino.stdSerializers.req,
+    res: (res) => ({
+      // FIXME: kind of a hack. body should be in in req. but have not being able to do it.
```

Comment:
we may want to fix this workaround before it's being in too many places in the codebase.
this issue arise before we were using apollo server. We were using another server before (don't recall the name right now) and it's possible now with apollo server this is not an issue any more

Reply:
Not sure about this one. I can't determine if it's still used.